tensorflow/lite/kernels/internal/reference/quantize.h

```diff
@@ -49,6 +50,39 @@ inline void AffineQuantize(const tflite::QuantizationParams& op_params,
   }
 }
 
+// Quantizes per-channel.
+template <typename InputT, typename OutputT>
+inline void PerChannelQuantize(
+    const tflite::PerChannelQuantizationParams& op_params,
+    const RuntimeShape& input_shape, const InputT* input_data,
+    const RuntimeShape& output_shape, OutputT* output_data) {
+  // Ensure flat size is same.
+  MatchingFlatSize(input_shape, output_shape);
+
+  const int32_t* zero_point = op_params.zero_point;
+  const float* scale = op_params.scale;
+  const int32_t quantized_dimension = op_params.quantized_dimension;
+  const int32_t num_dims = input_shape.DimensionsCount();
+  const int32_t* dims_data = input_shape.DimsData();
+  std::vector<int> current_dim(num_dims, 0);
```

Comment:
I think the usage of vector will introduce the unwanted "new" operation in tflite micro.  However, I also see "vector" is being used in tensorflow/lite/kernels/internal/reference/dequantize.h.  So I think this should be fine as long as  TFLite micro does not use this.

I would like opinion from @advaitjain or @petewarden to confirm.

Reply:
confirmed offline.